platform/spire/src/command.py

```diff
@@ -54,69 +39,136 @@ def configure(command: list, parser: argparse.ArgumentParser):
     return desc, configure
 
 
-def wrap(desc: str, func, param_tx=None):
-    sig = inspect.signature(func)
+class Command:
+    def __init__(self, func):
+        self.func = func
+        self.sig = inspect.signature(self.func)
+        self._remove_ops_from_sig()
+        self._command = None
+
+    def _remove_ops_from_sig(self):
+        parameters = list(self.sig.parameters.values())
+        if parameters[0].name != 'ops':
+            raise ValueError('first argument to command must be ops')
+        parameters = parameters[1:]
+        self.sig = self.sig.replace(parameters=parameters)
+
+    # so that this can still be called as the original function
+    def __call__(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
+
+    def operate(self, op, *args, **kwargs):
+        "Schedule this command to be run by Operations"
+        self.func(op, *args, **kwargs)
+
+    def process_args(self, cli_args):
+        "Process command-line arguments into function arguments"
+        # cli_args is the arguments object produced by argparse
+        cli_args = vars(cli_args)
```

Comment:
I think I just have a deeply held, unique and fundamentally incompatible stance on this issue from other people, since it just came up at work too.  _sigh_.  Yes, I will change the variable name :stuck_out_tongue: